code/app/commodity/server.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/brahma-adshonor/microservice/code/app/commodity/model"
+	"gorm.io/driver/mysql"
+	"gorm.io/gorm"
+)
+
+var (
+	db  *gorm.DB
+	err error
+)
+
+func hello(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintf(w, "hello\n")
+}
+func getProduct(w http.ResponseWriter, r *http.Request) {
+	var request ProductRequst
+	var response ProductResponse
+
+	fmt.Printf("http body is : %v\n", r.Body)
+	err := json.NewDecoder(r.Body).Decode(&request)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	fmt.Printf("ProductRequest is : %v\n", request)
+
+	var result *gorm.DB
+	if request.All {
+		result = db.Find(&response.Products)
+	} else {
+		result = db.Where(request.Ids).Find(&response.Products)
+	}
+
+	if result.Error != nil {
+		response.Err = result.Error
+	}
+	if result.RowsAffected == 0 {
+		response.Msg = "empty table"
+	}
+
+	fmt.Printf("Product Response is %v\n", response)
+	json.NewEncoder(w).Encode(&response)
+}
+
+func init() {
+	db, err = gorm.Open(mysql.New(mysql.Config{
```

Comment:
why not wrap it in a function like `New`, and keep `init` simple.

Reply:
first, the basic idea is we first build some toy servers could work under normal situations.
Then next chapter would be cold improved stage, like limitations and design